### PR TITLE
fix: Remove hard coded groups for spawners

### PR DESCRIPTION
**Pull request by EmosewaMC**
Tested that the target next to the plunger gun dude now spawns its target without needing to leave the area if it is spawned
Tested that spiders still spawn in Spider Queen battle
Tested that macros work

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]
thank you father for cleansing our sins

File: dScripts/02_server/Map/AG/NpcCowboyServer.cpp [hunk 1]
```diff
@@ -23,4 +24,17 @@ void NpcCowboyServer::OnMissionDialogueOK(Entity* self, Entity* target, int miss
...
 	}
+
+	// Next up hide or show the samples based on the mission state
+	int32_t visible = 1;
+	if (missionState == eMissionState::READY_TO_COMPLETE || missionState == eMissionState::COMPLETE_READY_TO_COMPLETE) {
+		visible = 0;
+	}
+
+	auto spawners = Game::zoneManager->GetSpawnersByName("PlungerGunTargets");
+	for (auto* spawner : spawners) {
```
Comment (jadebenn) [inline dScripts/02_server/Map/AG/NpcCowboyServer.cpp:35]: Nitpick: Does auto* spawner need to be mutable here?

Comment (EmosewaMC) [reply dScripts/02_server/Map/AG/NpcCowboyServer.cpp:35]: ehh i get it but its overkill tbqh

Comment (jadebenn) [reply dScripts/02_server/Map/AG/NpcCowboyServer.cpp:35]: tbf I am very much in the 'const all the things!' school of thought. Shows you whether your access is read/only or not and makes debugging easier. But I won't block if you really don't want to do it.

**Review by jadebenn — APPROVED** [review]

File: dScripts/02_server/Map/AG/NpcCowboyServer.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dScripts/02_server/Map/AG/NpcCowboyServer.cpp:35]: I get the mindset and will keep it in mind

**Pull request merged**